src/agentscope/service/web_search/web_digest.py

```diff
@@ -0,0 +1,298 @@
+# -*- coding: utf-8 -*-
+"""parsing and digesting the web pages"""
+import typing
+import json
+import re
+from typing import Optional, Callable, Literal, Any, Sequence
+import requests
+from loguru import logger
+
+
+from agentscope.service.service_response import ServiceResponse
+from agentscope.service.service_status import ServiceExecStatus
+from agentscope.models.model import ModelWrapperBase
+from agentscope.service import summarization
+
+
+DEFAULT_SYS_PROMPT = (
+    "You're a web page analyser. You job is to extract important"
+    "and useful information from html or webpage description.\n"
+)
+
+
+HTML_PARSING_TYPES = Literal[
+    "raw",
+    "selected_tags_to_text",
+    "self_define_func",
+]
+
+
+def is_valid_url(url: str) -> bool:
+    """
+    Use regex to check if a URL is valid
+    Args:
+        url (str): string to be checked
+
+    Returns:
+        bool: True if url is valid, False otherwise
+    """
+    # This regex pattern is designed to match most common URLs.
+    regex = re.compile(
+        r"^(https?://)"  # http:// or https://
+        r"(?:(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+"
+        r"[A-Z]{2,6}\.?|"  # domain...
+        r"localhost|"  # localhost...
+        r"\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})"  # ...or ip
+        r"(?::\d+)?"  # optional port
+        r"(?:/?|[/?]\S+)$",
+        re.IGNORECASE,
+    )  # rest of url
+    return re.match(regex, url) is not None
+
+
+def web_load(
+    url: str,
+    html_parsing_types: Sequence[HTML_PARSING_TYPES] = tuple(
```

Comment:
It would be better to make these tags into bool type switches

Reply:
Updated to controlled by three parameters:
* `keep_raw`: if true, ServiceResponse.content contains raw html str
* `html_selected_tags`: if is not None nor empty, ServiceResponse.content contains the text extracted from those tags.
* `html_parse_func`: if not None, ServiceResponse.content has the return processed by the `html_parse_func`. `html_parse_func` is expected to take the html str as input.